R/extremes-data.R

```diff
@@ -240,16 +240,16 @@ createDataRows <-
           if(!isUpchain){
             #Keep Maximum or Minimum based on current param
             if(param == "max"){
-              duplicateRows <- dataRows[order(dataRows$date, dataRows$related, dataRows$time, decreasing = TRUE),]
+              duplicateRows <- dataRows %>% arrange(desc(related)) %>% group_by(related) %>% arrange(date, time) %>% ungroup() %>% as.data.frame()
             } else  if (param == "min"){
-              duplicateRows <- dataRows[order(dataRows$date, dataRows$related, dataRows$time, decreasing = FALSE),]
+              duplicateRows <- dataRows %>% arrange(related) %>% group_by(related) %>% arrange(date, time) %>% ungroup() %>% as.data.frame()
             }
           } else {
             #Keep Maximum or Minimum based on current param
             if(param == "max"){
-              duplicateRows <- dataRows[order(dataRows$date, dataRows$primary, dataRows$time, decreasing = TRUE),]
+              duplicateRows <- dataRows %>% arrange(desc(primary)) %>% group_by(primary) %>% arrange(date, time) %>% ungroup() %>% as.data.frame()
             } else  if (param == "min"){
-              duplicateRows <- dataRows[order(dataRows$date, dataRows$primary, dataRows$time, decreasing = FALSE),]
+              duplicateRows <- dataRows %>% arrange(primary) %>% group_by(primary) %>% arrange(date, time) %>% ungroup() %>% as.data.frame()
```

Comment:
Ah okay, gotcha. I wasn't sure how to put it all into one call since the second arrange needs to happen within the groups created by the group_by. 

Also not sure why that test is failing on Travis. Just ran check locally and everything passed... 😕